docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,778 @@
+     Since {# nestable detached modifiers} can only contain a {# paragraphs}[paragraph] this can
+     cause severe limitations because insertion of (for example) code blocks is not possible. To
+     alleviate this deficiency, the {# indent segment extension} exists, which increases the
+     indentation of the tree by one level allowing nesting of more complex elements to occur.
+     The syntax is the following:
+     - The indent segment extension starts with a single `\` char
+     - This must be _immediately_ followed by a {# line ending}
+     -- Thus, the start node of this extension can be seen as a {# trailing modifier}
+     - The contents follow
+     - The extension ends when:
+     -- Another {# nestable detached modifier} occurs
+     -- Or when the indentation is manually closed with one of the {# delimiting modifiers}
+      +TODO(vhyrro): discuss what truly "automatically" ends the indent segment. I propose that the only way
+      to auto-close an indent segment is to use a structural detached modifier or a modifier of the same
+      type as the parent of the indent segment; i.e. if you attach an indent segment to a level 1
+      unordered list item then creating another level 1 unordered list item auto-closes the indent
+      segment. However, creating a level 1 ordered list segment inside would be okay.+
+
+     @code norg
+     - \
+       The indent segment extension has started.
+
+       I can now include multiple paragraphs within this list item.
+
+       @code lua
+       print("Even code blocks are allowed!")
+       @end
+
+       ^ Range-able detached modifiers can be inserted, too.
+
+       ---
+     - Another nestable detached modifier ends the indent segment
+     - However, in this example I already closed the indent segment manually with `---`
+     @end
+
+** Tags
+
+*** Carryover Tags
+    @code norg
+    |carryover param1 param2 "multi word param" "another param"
+    @end
+
+*** Ranged Tags
+
+**** Ranged Tags which support nested _Norg_ syntax
+     @code norg
+     #table
+     *heading* | /heading/ | _heading_
+     first | second | third
+     | 2 |
+     #end
+     @end
+
+**** Ranged Tags which do *not* support nested _Norg_ syntax (verbatim)
+     @code java
+     @MyAnnotation(name="someName", value="Hello World")
+     public class TheClass {
+       // ...
+     }
+     @end
+
+** Attached Modifiers
+   We can now discuss attached modifiers (which originally gave rise to the name of
+   {# detached modifiers} as their natural counter-parts). Here are the general rules for attached
+   modifiers:
+   - An opening modifier may only be preceded by {# whitespace} or {# punctuation}
+   - An opening modifier may _NOT_ be succeeded by {# whitespace}
+   - A closing modifier may _NOT_ be preceded by {# whitespace}
+   - A closing modifier may only be succeeded by {# whitespace} or {# punctuation}
+   - {# Attached modifiers} can only span at maximum a single {# paragraphs}[paragraph].
```

Comment:
I don't fully get this statement? Didn't we say we don't want any new lines supported inside attached modifiers? If not, then what does `at maximum a single paragraph` mean? Do you mean that new lines are okay, but `paragraph_break`s are not? If yes, this should be more explicit.

Reply:
When we were discussing ranged attached modifiers I said that I'd not be a fan of allowing paragraph breaks within them, but I did say that allowing regular newlines within attached modifiers was okay.

By "spans at maximum a single paragraph", I meant:
```
*this

is not allowed*
```
However:
```
*this
is fine*
```

However you're right in saying that it may be a little confusing to someone reading it without a proper example. I'll change that.